### PR TITLE
Adds the Account Registration Machine.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request

Atomized out of the previous econ balance PR to help keep things separate because I'm honestly not sold on my implementation here so I'll leave it here instead.

This adds the Account Registration Machine to the game, it's obtained at the computerized recordkeeping tech (Rather far in to the tech tree) and only obtained from the security lathe to prevent access from being too widespread, because I don't trust players have access to this until at LEAST half the shift has gone by. What this enables you to do is to create new accounts for blank IDs. The accounts won't have a parent payment account until actually assigned a job by a card console, but for all intents and purposes they'll act and interact like a standard, station bank account. 

## Why It's Good For The Game

There is literally no way to get a bank account if you're entering the round late, outside of the shuttle. This extends to savings accounts where you just want to hold money, or non-crew entering the station for the first time. If you want a new account, you're out of luck until you can mug an assistant  who went SSD prior to your arrival.

This will also make testing economy significantly easier off of live, as the only way to create multiple bank accounts for any purpose as a single player is re-spawning onto the server and creating a local set of 50-60 identical clones, which is hell and it's easier to just join a round in progress as an admin.

## Changelog
:cl:
add: Added the Account Registration Machine to the computerized record-keeping node, from the security lathe. 
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
make this be a public machine

**Comment by necromanceranne** [comment]
Could you stick one in the Op and Wizard ships?

**Comment by Fikou** [comment]
make it so hop console can do it too

**Comment by ArcaneMusic** [comment]
I am absolutely not of the mindset or interest to make spawning new bank accounts a hassle, due to how job assignments and payouts work.

File: code/game/machinery/accounting.dm [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+		if(!user.transferItemToLoc(I,src) || !anchored)
+			to_chat(user, "<span class='warning'>\the [src] blinks red as you try to insert the ID Card!</span>")
+			return
+		inserted_id = new_id
+		var/datum/bank_account/bank_account = new /datum/bank_account(inserted_id.registered_name, null)
+		inserted_id.registered_account = bank_account
+		playsound(loc, 'sound/machines/synth_yes.ogg', 30 , TRUE)
+		update_icon()
+		return
+	. = ..()
```
Comment (Rohesie) [inline code/game/machinery/accounting.dm:29]:
```suggestion
	return ..()
```

File: code/game/machinery/accounting.dm [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+/obj/machinery/accounting/attackby(obj/item/I, mob/living/user, params)
+	if(isidcard(I))
+		var/obj/item/card/id/new_id = I
+		if(inserted_id)
+			to_chat(user, "<span class='warning'>[src] already has a card inserted!</span>")
+			return
+		if(new_id.registered_account)
+			to_chat(user, "<span class='warning'>[src] already has a bank account!</span>")
+			return
+		if(!user.transferItemToLoc(I,src) || !anchored)
```
Comment (Rohesie) [inline code/game/machinery/accounting.dm:20]:
```suggestion
		if(!anchored || !user.transferItemToLoc(I,src))
```
You could lose the ID if it's not anchored that way

File: code/game/machinery/accounting.dm [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+			to_chat(user, "<span class='warning'>\the [src] blinks red as you try to insert the ID Card!</span>")
+			return
+		inserted_id = new_id
+		var/datum/bank_account/bank_account = new /datum/bank_account(inserted_id.registered_name, null)
```
Comment (Rohesie) [inline code/game/machinery/accounting.dm:24]:
```suggestion
		var/datum/bank_account/bank_account = new /datum/bank_account(inserted_id.registered_name)
```
not passing an arg is the same as passing `null`

File: code/game/machinery/accounting.dm [hunk 4]
```diff
@@ -0,0 +1,63 @@
+/obj/machinery/accounting
+	name = "account registration device"
+	desc = "A machine that allows heads of staff to create a new bank account after inserting an ID."
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "recharger"
+	circuit = /obj/item/circuitboard/machine/accounting
+	pass_flags = PASSTABLE
+	req_one_access = list(ACCESS_HEADS, ACCESS_CHANGE_IDS)
+	var/obj/item/card/id/inserted_id
```
Comment (Rohesie) [inline code/game/machinery/accounting.dm:9]: clean this variable both on `Destroy()` and also register to it getting deleted to clear it

**Comment by Rohesie** [comment]
Test it locally and comment here. If all is good, it's ready for merge.

**Comment by ArcaneMusic** [comment]
@Rohesie Should be good now, one small blip got left in that was a quick fix.

File: code/game/machinery/accounting.dm [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+/obj/machinery/accounting/Destroy()
+	. = ..()
+	if(inserted_id)
+		remove_card()
```
Comment (Rohesie) [inline code/game/machinery/accounting.dm:14]: Hell, I missed this one.
`Destroy()` should `return ..()` at the end of it, not call parent before.

**Pull request merged**